### ISSUE TITLE
False positives `unsupported-assignment-operation` and `unsubscriptable-object` for `TypedDict`s

### ISSUE THREAD

**Issue by christophe-riolo** [PyCQA/pylint#4610]
### Steps to reproduce

Given a file `test.py`:
```python
# pylint: disable=missing-docstring, pointless-statement
from typing import TypedDict


class MyDict(TypedDict):
    sample: int


my_dict = MyDict(sample=1)

my_dict['sample']
my_dict['sample'] = 2

```

### Current behavior

Result of `pylint test.py`:
```
************* Module test
test.py:11:0: E1136: Value 'my_dict' is unsubscriptable (unsubscriptable-object)
test.py:12:0: E1137: 'my_dict' does not support item assignment (unsupported-assignment-operation)
```

### Expected behavior

It should be passing.

### pylint --version output

Result of `pylint --version` output:

```
pylint 3.0.0a3
astroid 2.6.0
Python 3.9.5 (default, May 24 2021, 12:50:35) 
[GCC 11.1.0]

```

**Comment by Pierre-Sassoulas** [comment]
This is fixed in master, you can try using 2.8.3 or it will be released in 2.9.0 later on.

**Comment by Pierre-Sassoulas** [comment]
My bad, this is still affecting python 3.9 and 3.10.

**Comment by cdce8p** [comment]
@christophe-riolo Just wanted to let you know: The fix has already been merged in `astroid` and will be available with the next release.
